kuegi_bot/bots/strategies/kuegi_strat.py

```diff
@@ -34,29 +35,43 @@ def init(self, bars: List[Bar], account: Account, symbol: Symbol):
                           self.cancel_on_filter))
         super().init(bars, account, symbol)
 
-    def position_got_opened(self, position: Position, bars: List[Bar], account: Account, open_positions):
+    def position_got_opened_or_changed(self, position: Position, bars: List[Bar], account: Account, open_positions):
         other_id = TradingBot.get_other_direction_id(position.id)
         if other_id in open_positions.keys():
             open_positions[other_id].markForCancel = bars[0].tstamp
 
         # add stop
         gotStop= False # safety check needed to not add multiple SL in case of an error
+        gotTp = False
         for order in account.open_orders:
             orderType = TradingBot.order_type_from_order_id(order.id)
             posId = TradingBot.position_id_from_order_id(order.id)
             if orderType == OrderType.SL and posId == position.id:
                 gotStop= True
-                break
+                if abs(order.amount + position.current_open_amount) > self.symbol.lotSize / 2:
+                    order.amount = -position.current_open_amount
+                    self.order_interface.update_order(order)
+            elif self.tp_fac > 0 and orderType == OrderType.TP and posId == position.id:
+                gotTp = True
+                amount = self.symbol.normalizeSize(-position.current_open_amount + order.executed_amount)
+                if abs(order.amount - amount) > self.symbol.lotSize / 2:
+                    order.amount = amount
+                    self.order_interface.update_order(order)
+
         if not gotStop:
             order = Order(orderId=TradingBot.generate_order_id(positionId=position.id,
                                                            type=OrderType.SL),
                       stop=position.initial_stop,
                       amount=-position.amount)
             self.order_interface.send_order(order)
-            # added temporarily, cause sync with open orders is in the next loop and otherwise the orders vs
-            # position check fails
-            if order not in account.open_orders:  # outside world might have already added it
-                account.open_orders.append(order)
```

Comment:
is this removal intended? maybe some unwanted change from PR #15 ?

Reply:
intented, as this safety check is now done in trade_engine directly. -> no need to do it in every strat anymore